dspace-api/src/main/java/org/dspace/service/impl/ClientInfoServiceImpl.java

```diff
@@ -74,40 +73,86 @@ public String getClientIp(String remoteIp, String xForwardedForHeaderValue) {
     public boolean isUseProxiesEnabled() {
         if (useProxiesEnabled == null) {
             useProxiesEnabled = configurationService.getBooleanProperty("useProxies", true);
-            log.info("useProxies=" + useProxiesEnabled);
+            log.info("Proxies (useProxies) enabled? " + useProxiesEnabled);
         }
 
         return useProxiesEnabled;
     }
 
-    private IPTable parseTrustedProxyRanges(String[] proxyProperty) {
-        if (ArrayUtils.isEmpty(proxyProperty)) {
-            return null;
-        } else {
-            //Load all supplied proxy IP ranges into the IP table
-            IPTable ipTable = new IPTable();
+    /**
+     * Parse / Determine trusted proxies based on configuration. "Trusted" proxies are the IP addresses from which we'll
+     * allow the X-FORWARDED-FOR header. We don't accept that header from any IP address, as the header could be used
+     * to spoof/fake your IP address.
+     * <P>
+     * If "proxies.trusted.ipranges" configuration is specified, we trust ONLY those IP addresses.
+     * <P>
+     * If "proxies.trusted.ipranges" is UNSPECIFIED, we only trust the IP address(es) associated with ${dspace.ui.url}.
+     * This is necessary to allow the Angular UI server-side rendering (SSR) to send us the X-FORWARDED-FOR header,
+     * which it usually uses to specify the original client IP address.
+     * @return IPTable of trusted IP addresses/ranges, or null if none could be found.
+     */
+    private IPTable parseTrustedProxyRanges() {
+        IPTable ipTable = null;
+        // Whether we had to look up the UI's IP address (based on its URL), or not
+        boolean uiUrlLookup = false;
+
+        String[] ipAddresses = configurationService.getArrayProperty("proxies.trusted.ipranges");
+        String uiUrl = configurationService.getProperty("dspace.ui.url");
+
+        // If configuration is empty, determine IPs of ${dspace.ui.url} as the default trusted proxy
+        if (ArrayUtils.isEmpty(ipAddresses)) {
```

Comment:
I'd prefer to always add the UI url, even if proxies.trusted.ipranges is set (or to have a boolean property to set dspace.ui.url as not-trusted explicitly, which is disabled by default). This should reduce the amount of people accidentally breaking the UI integration just because they're adding another proxy to the config

Reply:
I agree with @benbosman , I would suggest to use `proxies.trusted.guess-ui-ip = true` as parameter for that

Reply:
Thanks for the feedback @benbosman  and @abollini . I've added a new `proxies.trusted.include_ui_ip = true` property which defaults to `true` & updated the code with tests to prove it works.